perf_bench_test.go

```diff
@@ -220,21 +271,59 @@ func serveVM(t *testing.T, vmGroup *sync.WaitGroup, vmID, requestID int, imageNa
 	}
 }
 
-// Control start and end of profiling
+// Goroutine function: Control start and end of profiling
 func profileControl(vmNum int, isProfile *bool, profiler *profile.Profiler) {
 	time.Sleep(time.Duration(*warmUpTime) * time.Second)
 	*isProfile = true
-	profiler.SetWarmTime()
+	profiler.SetWarmUpTime()
 	log.Info("Profile started")
 	time.Sleep(time.Duration(*profileTime) * time.Second)
 	*isProfile = false
-	profiler.SetTearDownTime()
+	profiler.SetCoolDownTime()
 	log.Info("Profile finished")
 }
 
+func latencyMeasurement(t *testing.T, vmID int, imageName string, isProfile *bool, results chan float64) {
+	var (
+		latencies  []float64
+		vmGroup    sync.WaitGroup
+		ticker     = time.NewTicker(500 * time.Millisecond)
+		vmIDString = strconv.Itoa(vmID)
+	)
+
+	for {
+		if tickerT := <-ticker.C; *isProfile && !tickerT.IsZero() {
+			vmGroup.Add(1)
+			go func() {
+				defer vmGroup.Done()
+				tStart := time.Now()
+				resp, _, err := funcPool.Serve(context.Background(), vmIDString, imageName, "replay")
+				require.Equal(t, resp.IsColdStart, false)
+				if err != nil {
+					log.Debugf("VM %s: Function returned error %v", vmIDString, err)
+				} else if resp.Payload != "Hello, replay_response!" {
+					log.Debugf("Function returned invalid: %s", resp.Payload)
+				}
+				latencies = append(latencies, float64(time.Since(tStart).Milliseconds()))
+			}()
+		} else if !*isProfile && len(latencies) > 0 {
+			vmGroup.Wait()
+			data := stats.LoadRawData(latencies)
+			mean, err := stats.Mean(data)
+			require.NoError(t, err, "Compute mean returned error")
+			percentile, err := stats.Percentile(data, 90)
+			require.NoError(t, err, "Compute percentile returned error")
+			results <- mean
```

Comment:
instead of returning 2 integers, create a type `LatencyStat` with two integer fields and return that one.

Reply:
Done